### ISSUE TITLE
High memory usage by radarr in comparison to sonarr

### ISSUE THREAD

**Issue by ShanDestromp** [Radarr/Radarr#158]
I have both an instance of sonarr and radarr running on the same virtual machine.  Said VM is a basic debian container under ProxMox; and is allocated 8GB of RAM (I had to increase this from 4 which Sonarr ran under easily); as well as allocating 16 CPU's (up from 8). (cat /proc/cpuinfo snippit below).

My Sonarr install has 100 listed series, 52 of which are monitored, and controls 7290 episode files.  Said install of Sonarr "idles" with 1.1% CPU and 2.8% memory usage.

In comparison in Radarr I have 615 movies, only 117 are monitored and yet the "non swapped" memory is 8 times larger, idling at 2.2cpu and 19.3% RAM (1621320 for Radarr vs 236544 Sonarr)

SNIPPIT FROM `PS`
`USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND`
`master   31843  2.2 19.3 2554464 1621320 ?     Ssl  Jan10  16:33 /usr/bin/mono /opt/radarr/Radarr.exe -nobrowser`
`master   32076  1.1  2.8 2609380 236544 ?      Ssl  00:44   5:39 /usr/bin/mono /opt/NzbDrone/NzbDrone.exe -nobrowser`

CPUINFO

`processor       : 31
vendor_id       : GenuineIntel
cpu family      : 6
model           : 45
model name      : Intel(R) Xeon(R) CPU E5-2650 0 @ 2.00GHz
stepping        : 7
microcode       : 0x710
cpu MHz         : 2474.062
cache size      : 20480 KB
physical id     : 1
siblings        : 16
core id         : 7
cpu cores       : 8
apicid          : 47
initial apicid  : 47
fpu             : yes
fpu_exception   : yes
cpuid level     : 13
wp              : yes
flags           : fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc aperfmperf eagerfpu pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic popcnt tsc_deadline_timer aes xsave avx lahf_lm epb tpr_shadow vnmi flexpriority ept vpid xsaveopt dtherm ida arat pln pts
bugs            :
bogomips        : 4001.47
clflush size    : 64
cache_alignment : 64
address sizes   : 46 bits physical, 48 bits virtual
power management:
`

**Comment by fedoranimus** [comment]
This is not unexpected. Much of the Sonarr processes are still running, but are unused. We have no done any work on streamlining Radarr.

Thank you for sharing this, it helps us understand how Radarr performs. We'll dig performance in a later release.

**Comment by ShanDestromp** [comment]
I mentioned it in another issue thread (import all)  but I believe this high memory usage contributes to the overall system lag.  Radarr was relatively snappy before I imported all my movies, but as I added movies to the system and memory spiked the UI became very sluggish.

**Comment by onedr0p** [comment]
@ShanDestromp I never would trust CP with all my movies. I only used it for grabbing releases & and auto importing from a feed, not to say this is the aim of Radarr, but even Sonarr slows down with 300+ series.

**Comment by fedoranimus** [comment]
It sounds mostly like a scaling issue; it may not be solvable until a critical look at the UI is taken (which I'm digging into).

**Comment by ShanDestromp** [comment]
@onedr0p The reason I filed a bug report is that while I have more movies (compared to TV series in Sonarr); I have significantly more data that Sonarr is handling.  Almost 8000 episodes vs 600 movies.

If I have Sonarr display my listing for Doctor Who, which has over 700 episodes (in 696 files + metadata, images etc) split among 26 seasons the UI is *far* more responsive than Radarr displaying roughly the same amount of data for my movie collection.

Again, I'm not complaining, Radarr has already proven itself better for my needs than CP (radarr has found releases I've been waiting on for weeks that CP never sees apparently); I'm just (trying) to provide as much data as possible to @fedoranimus and @galli-leo to help "solve" it - if it is indeed a bug.  As @fedoranimus says, it could just be an issue of scale.

With that said, my instance of CP (which again, maintains my whole collection) still uses less resources than Radarr (similar usage to Sonarr with my TV).

**Comment by ShanDestromp** [comment]
@fedoranimus @galli-leo I've been keeping an eye on my system usage since I initially reported this issue and to date it does not seem to have re-occurred.  Not sure what actually fixed it or if it was just an artifact of the conversion from TV to movies or something else unique to my system but either way I think it's safe to consider it closed for now.  I do still get UI lag viewing my full movie list but as @onedr0p noted that's not specifically a Radarr issue but one Sonarr has as well.

**Comment by kanenses** [comment]
I have the same issue!

**Comment by Doonut** [comment]
same here, over 3x the usage of sonarr

**Comment by galli-leo** [comment]
How many movies do you guys have?

**Comment by galli-leo** [comment]
Also what is the exact amount of Ram consumed by radarr?

**Comment by kanenses** [comment]
i have around 19 movies, and a consume around 1.5gb RAM :-(

**Comment by ShanDestromp** [comment]
Since I was the original reporter (and closer) of this issue I figured I'd post an update since others seem to still be having issues.

My Radarr install still uses more than Sonarr however it is drastically less than when I originally reported things.  Still on the same hardware and VM settings as original:


`master     276  0.6  6.7 2450924 563420 ?      Ssl  Sep01   9:02 /usr/bin/mono /opt/radarr/Radarr.exe -nobrowser
master   21929  0.4  1.9 1834080 162660 ?      Ssl  18:46   0:42 /usr/bin/mono /opt/NzbDrone/NzbDrone.exe -nobrowser
`

`root@ferengi:~# free -h`
`              total        used        free      shared  buff/cache   available`
`Mem:           8.0G        852M        7.1G        150M         62M        7.1G`
`Swap:          8.0G          0B        8.0G`

As you can see while Radarr uses substantially more than Sonarr (by about 3x); we're still talking only around 580MB of active RAM for ~680 movies (30 of which are monitored).

For others thinking they might have an issue, and happen to be using `ps` to get memory usage remember that the VSZ column isn't the total "in use".  To quote someone elses analogy VSZ is the size of your plate, but RSS is how much food you've actually got on that plate.

**Comment by redec** [comment]
Hello, new radarr user here, veteran sonarr user. Is anything in the works for this yet? I just installed radarr about a week ago, haven't imported any movies into it, I just added like 10 wanted movies to it, and its memory usage is ridiculous. My sonarr has maybe 50 shows managing a thousand or so episodes, and its memory usage is a fraction of radarr's.  Radarr is using significantly more RAM (~5x) and significantly more CPU (~10x) and its really not doing much yet.  I dunno maybe (hopefully) I have something misconfigured. At any rate I've had to turn it off for now because it was running my machine out of memory.

**Comment by galli-leo** [comment]
@redec The latest pre-release should have some optimizations in there. Additionally, it is important to verify that Radarr is actually consuming that much memory or whether that is with files, etc. mapped taken into account. 

There is some good discussion about this, here: https://github.com/Radarr/Radarr/issues/1580#issuecomment-348690566

If you do upgrade your Radarr, try deleting your logs.db file, that seems to help.
